services/wallet/controllers_v3_test.go

```diff
@@ -690,6 +690,81 @@ func TestLinkGeminiWalletV3FirstLinking(t *testing.T) {
 	}
 }
 
+func TestLinkZebPayWalletV3_InvalidKyc(t *testing.T) {
+
```

Comment:
In #1956 I removed this test because claims have been refactored and covered with tests.

I assume that this test is being brought in back because it checks that the status code is what's expected. This is a good intent. However, possibly we could approach that from a different angle.

Specifically, we could craft test data such that we call `service.LinkZebPayWallet` and test the returned error to have the desired properties, specifically the status code.

I think that that would be enough to ensure our expectations are met, but in a more focussed manner.

Reply:
The main premise was to verify that we get the appropriate status code back from the controller which is why I brought this test back.  I believe there is also already a check to make sure we get the appropriate error back from the service as well.  Do you suggest we remove this test again, it was put in due to the defect that the controller was not returning the appropriate status code.

Reply:
I see, thanks for the clarification. I don't think there currently are tests for the service methods (e.g. for the linking methods). Ideally, we can and want to achieve the same outcome as this test, but by directly testing `LinkZebPayWallet`. That can be done later.

Let's keep this test for now. Like I said, it was not a blocking comment.